### PR TITLE
New Deposit UI + Zappers

### PR DESCRIPTION
**Pull request by mohsenParvar**

### CODE REVIEW

**Comment by prodvdc** [comment]
@mohsenParvar Mobile view needs some stuff for UX
image

**Comment by prodvdc** [comment]
@mohsenParvar on stable vault, it has an issue when selecting other token from combo box.
image

File: components/Icons/ArrowDownIcon/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+import { memo } from 'react'
+import SvgIcon from '@material-ui/core/SvgIcon'
+import { makeStyles } from '@material-ui/core/styles'
+import clsx from 'clsx'
+
+const useStyles = makeStyles(() => ({
+  root: {
```
Comment (saito-sv) [inline components/Icons/ArrowDownIcon/index.js]: Please remove or add a description.

**Review by saito-sv — CHANGES_REQUESTED** [review]

**Review by ghost — DISMISSED** [review]
Looks like vercel is having problems deploying your last commit. Can you look into it so I can test it properly?

File: parts/Compound/CompoundDialogs/CompoundInfo/index.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+  const tokensWithAmountToPut = tokensWithPriceAndAmount.map((item) => {
+    return ({
+      ...item,
+      amountToPut: divide(amountOfUsdToPut, item.price).toFixed(18)
+    })
+  })
+
+  const mixedTokensValue = divide(amountOfUsdToPut, pool.pricePoolToken)
+
```
Comment (saito-sv) [inline parts/Compound/CompoundDialogs/CompoundInfo/index.js]: remove unnecessary extra spaces.

File: parts/Compound/CompoundDialogs/index.js [hunk 3]
```diff
@@ -47,175 +66,232 @@ const useStyles = makeStyles((theme) => ({
...
+      }
+    })
+  }
+  useEffect(() => {
+    dispatch({ type: compoundDialogActionTypes.setUserData, payload: userData })
+    return () => {
+
+    }
+  }, [userData, userData.token0Balance])
+
```
Comment (saito-sv) [inline parts/Compound/CompoundDialogs/index.js:223]: remove extra spacing

**Review by saito-sv — DISMISSED** [review]

File: parts/Compound/CompoundListItem/DetailItem/index.js [hunk 4]
```diff
@@ -37,14 +37,14 @@ const DetailItem = ({ item, userBoost, totalAPY }) => {
...
+    <Grid
       container
       direction="column"
       display="flex"
     >
-        {item.deprecatedPool && <Typography align="center" color="error">
-          This pool is deprecated and doesn{`'`}t receive rewards anymore, please withdraw from it.
-        </Typography>}
+      {item.deprecatedPool && <Typography align="center" color="error">
+        This pool is deprecated and don{`'`}t receive rewards anymore, please withdraw from it.
```
Comment (ghost) [inline parts/Compound/CompoundListItem/DetailItem/index.js]: it was right before

File: parts/Compound/CompoundDialogs/reducer.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        }
+            break
+        case compoundDialogActionTypes.setSliderValue: {
+            const value = action.payload
+            const usedBalance = selectedToken.balance.mul(value).div(100)
+            if (selectedToken.balance && (BNToFloat(usedBalance) > 0 || value === 0)) {
+                const inputAmount = (usedBalance / 10 ** userData?.lpDecimals)
+                newState.mixedTokenValue = calculatedBalance({ userData, title, value })
+                newState.amount = usedBalance
+                newState.inputAmount = inputAmount > 1e-6 ? inputAmount : Number(inputAmount).toLocaleString('en-US', { maximumSignificantDigits: 18 })
```
Comment (ghost) [inline parts/Compound/CompoundDialogs/reducer.js]: needs a line break

File: contexts/compound-and-earn-context.js [hunk 6]
```diff
@@ -377,7 +401,72 @@ export function CompoundAndEarnProvider({ children }) {
...
+        //1% slippage
+        const amountMinToken = estimateSwap.swapAmountOut.sub(estimateSwap.swapAmountOut.div(100));
+        const zapTx = await zappersContract.zapInAVAX(item.address, amountMinToken, item[`token${tokenPos}`].address, {value: amount});
+        
+        return zapTx;
+      }
+    }
+  }
+
+  const deposit = async (item, amount, addressFromZapper, onlyGauge = false, isNativeAVAX = false) => {
```
Comment (ghost) [inline contexts/compound-and-earn-context.js]: We need to check if this function is not used elsewhere, because it's declaration changed

**Review by ghost — COMMENTED** [review]
I'm not very confident about the token0balance, token1balance and AVAXBalance autorefresh, we should test some edge cases:
- Get AVAX, token0 and token1 from TJ/Pangolin on a tab, try to deposit on snowball on the other tab without hard refreshing.

**Comment by ghost** [comment]
We should look for regressions too, the base of this PR is very old

**Comment by ghost** [comment]
> I'm not very confident about the token0balance, token1balance and AVAXBalance autorefresh, we should test some edge cases:
> 
> * Get AVAX, token0 and token1 from TJ/Pangolin on a tab, try to deposit on snowball on the other tab without hard refreshing.

tested, seems to be working fine

**Review by ghost — APPROVED** [review]
LGTM
